Add support for "packages" install key in v2 config (#7289)
readthedocs/readthedocs.org: docs/config-file/v2.rst

  manage the build, this setting will not have any effect. Instead
  add the extra requirements to the ``environment`` file of Conda.

> Packages list

Author: Done

'''''''''''''

> Install packages from a list of requirements. This allows you to embed requirements in the
> ``.readthedocs.yml`` configuration directly.

Reviewer: ```suggestion
Install packages from a list of requirements.
This allows you to declare dependencies in the ``.readthedocs.yml`` file instead of creating a separate file.
```

:Key: `packages`

> :Type: ``string``

Reviewer: ```suggestion
:Type: ``list``
:Default: ``[]``
```

:Required: ``true``

Example:

.. code-block:: yaml

   version: 2

   python:
     version: 3.7
     install:
       - packages:
         - mkdocs-material
         - Pygments

> 

Reviewer: We should add something like the packages section

`With the previous settings, Read the Docs will execute the next commands:`...

   pip install .[docs]
   python package/setup.py install

Extra dependencies
''''''''''''''''''

Install packages from a list of requirements.
This allows you to declare dependencies in the ``.readthedocs.yml`` file instead of creating a separate file.

:Key: `packages`
:Type: ``list``
:Default: ``[]``

Example:

.. code-block:: yaml

   version: 2

   python:
     version: 3.7
     install:
       - packages:
         - mkdocs-material

>          - Pygments

Reviewer 2: We should consider supporting private packages https://docs.readthedocs.io/en/stable/guides/private-python-packages.html for Read the Docs for Business, probably.

Author: Unless I am missing something, this should already be supported implicitly. You can just use a Git repository URL instead of a package name, e.g.:

```
- packages:
  - mkdocs-material
  - git+https://${GITHUB_TOKEN}@github.com/user/project.git@{version}
```

Reviewer 2: Have you tested that the environment variable is properly replaced with its value?

Author: Nope and I doubt that it will be. Good point that we should probably support the substituation of environment variables.

Does Pip support the substitution in `requirements.txt`?

Reviewer: > Does Pip support the substitution in requirements.txt?

Yes, pip supports expanding variables

Author: Maybe it'll be easier to just write the arguments into a temporary requirements file and re-use the existing logic for that

Reviewer 3: In order to push this along and finish this work, I'd be fine opening an issue to add private repository support to the `packages` config key later. This PR seems close otherwise and it looks to be blocked on this additional feature.

   pip install .[docs]
   python package/setup.py install

Extra dependencies
''''''''''''''''''

Install packages from a list of requirements.
This allows you to declare dependencies in the ``.readthedocs.yml`` file instead of creating a separate file.

:Key: `packages`
:Type: ``list``
:Default: ``[]``

> :Required: ``true``

Reviewer: It has a default, so isn't required

```suggestion
```
